### ISSUE TITLE
[feature request] dump Code Notes

### ISSUE THREAD

**Issue by meleu** [Jamiras/RATools#7]
While thinking on that feature I requested in #6 I thought that the input box for entering the game ID can also have a checkbox saying `dump Code Notes`.

The resulting output could be as in this example below (I intentionally got an example where some code notes have newlines):

```
// Zooming Secretary
// #ID = 4650
//
// CODE NOTES:
// - 0x0004: Input
// - 0x012e: 16bit fired = c569
// - 0x0336: Speed:
// Coffee = goes to 0x22 and gradually decreases to 0x18
// Normal = 0x18
// Chatting = decreases 0x8 from the current speed
// - 0x033a: Coffee effect counter (>0 means under coffee effect)
// ...
```

IMHO it would be extremelly useful to start coding as soon as possible.

Thanks.

**Comment by Jamiras** [comment]
I've integrated this functionality into the Dump Published dialog (now the New Script dialog). The right side of the dialog provides a list of addresses referenced by the selected achievements and their associated code notes. If no achievements are selected, all known addresses and their associated code notes are displayed.

For each address, you can define a function. Each defined function will be dumped to the script with the associated code notes:
```
// 2 Players - 00=2Players, 02=1Player
// First time it changes is at the character select screen
function current_players() => byte(0x00002E)
```

Notes will only be dumped for generated functions. If an function name is not provided for an address, the associated notes will not be dumped.

**Comment by meleu** [comment]
[I'm reopening this issue/feature request, but feel free to close if you don't agree.]

@Jamiras 
> Notes will only be dumped for generated functions. If an function name is not provided for an address, the associated notes will not be dumped.

This isn't what I expected when opened this feature request...

Let me describe my workflow:

1. research about the game looking for interesting cheevos (youtube, gamefaqs, etc.)
2. write down all cheevos ideas
3. RAM dig: look for the useful addresses to implement the cheevos. I invest some time in this step, looking for **all interesting addresses**.
4. **after finding the addresses** I start to write the script. This step is fast after knowing the address.
5. testing, badges, etc...

Summing up: when I start to actually write the script, most of the addresses already have notes. Of course I go back to RAM digging after starting to code the script, but the majority of the addresses are already known.

That's why I said that dumping all code notes on comments would be extremelly useful to start coding as soon as possible.

**Comment by Jamiras** [comment]
I have added two new combo boxes to the New Script dialog. The first allows you to specify whether the listed memory addresses should be filtered for just the selected achievements. The second allows you to specify whether code note comments should only be generated for memory addresses associated to functions.

The default selections maintain the existing functionality (only show memory addresses for selected achievements and only dump notes for memory addresses assigned function names).

If the script being created is associated to a game with no defined achievements, the filter will automatically change to show all known memory addresses. The second combo box will still be set to only dump notes for memory addresses assigned function names.

For your use case, you'll want to change both combo boxes to "All". The resulting output looks something like this:
```
// $0074: cursor X
// $0075: cursor Y
// $0078: [32-bit] score

// $007C: [16-bit] level (x5)
function current_level() => word(0x00007C)

// $0080: [16-bit] duration of game (number of times $51=$87)
// $0083: b0=water is fatal
//        b1=bottomless swamps
//        b2=cannot build
//        b3=only build up
//        b4=only build near towns
```

**Comment by meleu** [comment]
oh yeah! cool!
I'm looking forward to test it, are you planning to release a binary anytime soon?

**Comment by meleu** [comment]
**Woohool!!**

I've just tested it and it's exactly what I want! Many thanks @Jamiras !
